packages/big-design/src/components/Stepper/Stepper.tsx

```diff
@@ -0,0 +1,104 @@
+import { CheckIcon } from '@bigcommerce/big-design-icons';
+import React, { HTMLAttributes, memo, useEffect, useState } from 'react';
+
+import {
+  StyledDash,
+  StyledLight,
+  StyledLightText,
+  StyledMobileLight,
+  StyledMobileLightText,
+  StyledStep,
+  StyledStepper,
+  StyledText,
+} from './styled';
+
+export interface StepProps {
+  number: number;
+  text?: string;
+  state?: 'pending' | 'current' | 'done';
+  isLast?: boolean;
+}
+
+export interface StepperProps extends HTMLAttributes<HTMLDivElement> {
+  steps: Array<string>;
+  currentStep: number;
+}
+
+export interface LightProps {
+  active?: boolean;
+}
+export interface DashProps {
+  active?: boolean;
+}
+export interface TextProps {
+  active?: boolean;
+}
+
+export const Step: React.FC<StepProps> = ({ number, text, state, isLast }) => {
+  const [isCurrent, setIsCurrent] = useState(state === 'current');
+  const [isDone, setIsDone] = useState(state === 'done');
+  useEffect(() => {
+    setIsCurrent(state === 'current');
+    setIsDone(state === 'done');
+  }, [state]);
```

Comment:
I would say that this `Stepper` component is not a stateful component. We should avoid having internal state where it's not needed. This is more of a component where you pass state into it and you can derive values from it. See what I mean [here](https://github.com/Spawbreaker/big-design/compare/stepper...chanceaclark:stepper-updates#diff-0e9b4bcc8c6ecb6bd81b845339f2ec126e6b5eb7bcb5f8930fb47d7e5671981dR14-R16).